tests/WriteDefaultsTests.js

```diff
@@ -0,0 +1,204 @@
+                    }
+                }
+            }
+        },
+        sequence: [
+            {
+                func: "{readDefaultsRequest}.send"
+            },
+            {
+                event: "{readDefaultsRequest}.events.onComplete",
+                listener: "kettle.test.assertErrorResponse",
+                args: [{
+                    message: "Read Defaults returns 404 as we haven't created the new grade yet",
+                    errorTexts: "Grade not found",
+                    string: "{arguments}.0",
+                    request: "{readDefaultsRequest}",
+                    statusCode: 404
+                }]
+            },
+            {
+                func: "{writeDefaultsRequest}.send",
+                args: [gpii.tests.nexus.writeDefaults.newGradeOptions]
+            },
+            {
+                event: "{writeDefaultsRequest}.events.onComplete",
+                listener: "gpii.test.nexus.assertStatusCode",
+                args: ["{writeDefaultsRequest}", 200]
+            },
+            {
+                func: "{readDefaultsAgainRequest}.send"
+            },
+            {
+                event: "{readDefaultsAgainRequest}.events.onComplete",
+                listener: "gpii.test.nexus.verifyReadDefaultsResponse",
+                args: ["{arguments}.0", "{readDefaultsAgainRequest}", ["fluid.component", "gpii.tests.nexus.writeDefaults.newGrade"]]
+            }
+
+            // TODO: Update the grade definition and verify
+
+        ]
+    },
+    {
+        name: "Write Defaults with badly formed JSON",
+        gradeNames: "gpii.test.nexus.testCaseHolder",
+        expect: 3,
+        config: {
+            configName: "gpii.tests.nexus.config",
+            configPath: configPath
+        },
+        testGradeName: "gpii.tests.nexus.writeDefaults.badlyFormedJson",
+        sequence: [
+            {
+                func: "{writeDefaultsRequest}.send",
+                args: [
+                    "{",
+                    {
+                        headers: {
+                            "Content-Type": "application/json"
+                        }
+                    }
+                ]
+            },
+            {
+                event: "{writeDefaultsRequest}.events.onComplete",
+                listener: "kettle.test.assertErrorResponse",
+                args: [{
+                    message: "Write Defaults returns 400 for badly formed JSON",
+
+                    // TODO: Maybe "Bad request" rather than "Unknown error"
+
+                    // The "Unknown error" default message is set in
+                    // kettle.request.http.errorHandler and in this
+                    // case, the error originates from Express
+                    // body-parser middleware
+                    // (body-parser/lib/read.js). We should implement
+                    // a replacement JSON parsing middleware based on
+                    // kettle.JSON.parse(), with richer error
```

Comment:
I've filed 2 JIRA issues for this work:

* https://issues.gpii.net/browse/GPII-1541
* https://issues.gpii.net/browse/GPII-1769